File: mobile/lib/data/models/sport/training_history.dart
import 'package:mobile/data/models/sport/training_program.dart';

class TrainingHistory {
  final int id;
  final int userId;
  final int trainingProgramId;
  final int duration;
  final int? caloriesBurned;
  final double? weightBefore;
  final double? weightAfter;
  final DateTime createdAt;
  final TrainingProgram? trainingProgram;

  TrainingHistory({
    required this.id,
    required this.userId,
    required this.trainingProgramId,
    required this.duration,
    this.caloriesBurned,
    this.weightBefore,
    this.weightAfter,
    required this.createdAt,
    this.trainingProgram,
  });

  factory TrainingHistory.fromJson(Map<String, dynamic> json) {
    return TrainingHistory(
      id: json['id'],
      userId: json['user_id'],
      trainingProgramId: json['training_program_id'],
      duration: json['duration'],
      caloriesBurned: json['calories_burned'],
      weightBefore: (json['weight_before'] as num?)?.toDouble(),
      weightAfter: (json['weight_after'] as num?)?.toDouble(),
      createdAt: DateTime.parse(json['created_at']),
      trainingProgram:
          json['program'] != null
              ? TrainingProgram.fromJson(json['program'])
              : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'training_program_id': trainingProgramId,
      'duration': duration,
      'calories_burned': caloriesBurned,
      'weight_before': weightBefore,
      'weight_after': weightAfter,
    };
  }
}

File: backend/app/Models/TrainingHistory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class TrainingHistory extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'training_program_id',
        'duration',
        'weight_before',
        'weight_after',
        'date',
        'calories_burned',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function trainingProgram(): BelongsTo
    {
        return $this->belongsTo(TrainingProgram::class);
    }

    public function sport(): BelongsTo
    {
        return $this->belongsTo(Sport::class);
    }
}
